api_client/python/timesketch_api_client/scenario.py

```diff
@@ -103,19 +165,119 @@ def __init__(self, sketch_id, question_id, uuid, api):
         self.uuid = uuid
         self.api = api
         self.sketch_id = sketch_id
+        self._name = None
+        self._display_name = None
+        self._description = None
+        self._data = None
         super().__init__(
             api=api, resource_uri=f"sketches/{self.sketch_id}/questions/{self.id}/"
         )
 
+    def _update(self, data):
+        """Internal helper to update the question."""
+        resource_url = f"{self.api.api_root}/{self.resource_uri}"
+        response = self.api.session.post(resource_url, json=data)
+        self._data = error.get_response_json(response, logger)
+        updated_object = self._data["objects"][0]
+        self._name = updated_object["name"]
+        self._display_name = updated_object["display_name"]
+        self._description = updated_object["description"]
+
+    def add_attribute(self, name, value, ontology=None, description=None):
+        """Adds an attribute to the question.
+
+        Args:
+            name (str): The name of the attribute.
+            value (str): The value of the attribute.
+            ontology (str): Optional ontology for the attribute.
+            description (str): Optional description for the attribute.
+        """
+        attribute = {
+            "name": name,
+            "value": value,
+            "ontology": ontology,
+            "description": description,
+        }
+        self._update({"attributes": [attribute]})
```

Comment:
I think here is a logic flaw, you are not adding it but replace all attributes existing before with the to be added one?

roughly:
```
current_attributes = self._data.get("attributes", [])
current_attributes.append(new_attribute)
self._update({"attributes": current_attributes})
```